### PR TITLE
Add test case class for blueprints

### PR DESCRIPTION
**Pull request by phobologic**
This is a derivative of something @ejholmes created for us internally at Remind, and it's been really useful.

### CODE REVIEW

**Comment by phobologic** [comment]
Here's an example test:

```
import unittest

from stacker.blueprints.testutil import BlueprintTestCase
from stacker.context import Context
from stacker.variables import Variable

from stacker_blueprints.ecr import Repositories


class TestRepositories(BlueprintTestCase):
    def test_create_template(self):
        ctx = Context({'namespace': 'test'})
        blueprint = Repositories('test_repo', ctx)
        blueprint.resolve_variables(
            [
                Variable('Repositories', ["repo1", "repo2"])
            ]
        )
        blueprint.create_template()
        self.assertRenderedBlueprint(blueprint)


if __name__ == '__main__':
    unittest.main()
````

Here's an example output for the first time it's ran:

```
# python tests/test_ecr.py
E
======================================================================
ERROR: test_create_template (__main__.TestRepositories)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_ecr.py", line 20, in test_create_template
    self.assertRenderedBlueprint(blueprint)
  File "build/bdist.macosx-10.10-x86_64/egg/stacker/blueprints/testutil.py", line 29, in assertRenderedBlueprint
    with open(expected_output) as fd:
IOError: [Errno 2] No such file or directory: 'tests/fixtures/blueprints/test_repo.json'

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
```

It will result in a new file being written called `tests/fixtures/blueprints/test_repo.json-result`.  If you move that to `tests/fixtures/blueprints/test_repo.json`, then tests will pass:

```
# cat tests/fixtures/blueprints/test_repo.json-result
{
    "Resources": {
        "repo1Repository": {
            "Properties": {
                "RepositoryName": "repo1"
            },
            "Type": "AWS::ECR::Repository"
        },
        "repo2Repository": {
            "Properties": {
                "RepositoryName": "repo2"
            },
            "Type": "AWS::ECR::Repository"
        }
    }
}
# mv tests/fixtures/blueprints/test_repo.json-result tests/fixtures/blueprints/test_repo.json
# python tests/test_ecr.py
.
----------------------------------------------------------------------
Ran 1 test in 0.001s

OK
```

If the blueprints don't match (for example, I just added a bad value to the output file) this is the exception you'll get:

```
# echo "bad_value" >> tests/fixtures/blueprints/test_repo.json
# python tests/test_ecr.py
F
======================================================================
FAIL: test_create_template (__main__.TestRepositories)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/test_ecr.py", line 20, in test_create_template
    self.assertRenderedBlueprint(blueprint)
  File "build/bdist.macosx-10.10-x86_64/egg/stacker/blueprints/testutil.py", line 32, in assertRenderedBlueprint
    self.assertEqualsDiff(rendered, expected)
  File "build/bdist.macosx-10.10-x86_64/egg/stacker/blueprints/testutil.py", line 19, in assertEqualsDiff
    self.assertEquals(a, b, diff(a, b))
AssertionError:   {
      "Resources": {
          "repo1Repository": {
              "Properties": {
                  "RepositoryName": "repo1"
              },
              "Type": "AWS::ECR::Repository"
          },
          "repo2Repository": {
              "Properties": {
                  "RepositoryName": "repo2"
              },
              "Type": "AWS::ECR::Repository"
          }
      }
- }
+ }bad_value

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (failures=1)
```

File: stacker/tests/blueprints/test_testutil.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+class TestRepositories(BlueprintTestCase):
+    def test_create_template_passes(self):
+        ctx = Context({'namespace': 'test'})
+        blueprint = Repositories('test_repo', ctx)
+        blueprint.resolve_variables([
+            Variable('Repositories', ["repo1", "repo2"])
+        ])
+        blueprint.create_template()
+        self.assertRenderedBlueprint(blueprint, dump=False, expected=expected)
```
Comment (ejholmes) [inline stacker/tests/blueprints/test_testutil.py]: atm, this is really just testing `assertEqualsDiff` if we pass in `expected`. Any reason not to just let this read from the file?

File: stacker/blueprints/testutil.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    )
+
+
+class BlueprintTestCase(unittest.TestCase):
+    def assertEqualsDiff(self, a, b):  # noqa: N802
+        self.assertEquals(a, b, diff(a, b))
+
+    def assertRenderedBlueprint(self, blueprint, dump=True,  # noqa: N802
+                                expected=None):
+        expected_output = "tests/fixtures/blueprints/%s.json" % blueprint.name
```
Comment (ejholmes) [inline stacker/blueprints/testutil.py]: I feel like this path should probably be configurable, maybe with a class variable (is there a common "fixture path" for unittest?).

File: stacker/tests/blueprints/test_testutil.py (hunk 1 above)
Comment (phobologic) [inline stacker/tests/blueprints/test_testutil.py]: Good call, updated.

File: stacker/blueprints/testutil.py (hunk 2 above)
Comment (phobologic) [inline stacker/blueprints/testutil.py]: Good idea, updated!

**Pull request merged**